main.go

```diff
@@ -12,29 +16,81 @@ const (
 // generateRandomElements generates random elements.
 func generateRandomElements(size int) []int {
 	// ваш код здесь
+	if size <= 0 {
+		return []int{}
+	}
+
+	result := make([]int, size)
+	for i := range result {
+		result[i] = rand.Int()
+	}
+
+	return result
 }
 
 // maximum returns the maximum number of elements.
 func maximum(data []int) int {
-	// ваш код здесь
+	if len(data) == 0 {
+		return 0
+	}
+	if len(data) == 1 {
+		return data[0]
+	}
+	return slices.Max(data)
+
 }
 
 // maxChunks returns the maximum number of elements in a chunks.
 func maxChunks(data []int) int {
 	// ваш код здесь
+	if len(data) == 0 {
+		return 0
+	}
+	if len(data) == 1 {
+		return data[0]
+	}
+
+	var wg sync.WaitGroup
+	size := len(data) / CHUNKS
+	maxCounts := make([]int, CHUNKS)
+
+	wg.Add(CHUNKS)
+
+	for i := 0; i < CHUNKS; i++ {
+		sliceFirst := i * size
+		sliceEnd := sliceFirst + size
+
+		go func() {
+			defer wg.Done()
+			maxCounts[i] = slices.Max(data[sliceFirst:sliceEnd])
+		}()
+	}
+	wg.Wait()
+	return slices.Max(maxCounts)
 }
 
 func main() {
 	fmt.Printf("Генерируем %d целых чисел", SIZE)
-	// ваш код здесь
+	data := generateRandomElements(SIZE)
 
 	fmt.Println("Ищем максимальное значение в один поток")
 	// ваш код здесь
+	var (
+		start   time.Time
+		max     int
+		elapsed time.Duration
+	)
+
+	start = time.Now()
+	max = maximum(data)
+	elapsed = time.Duration(time.Since(start).Microseconds())
```

Comment:
Нет смысла оборачивать
```suggestion
	elapsed = time.Since(start).Microseconds()
```

Reply:
У меня компилятор сам его оборачивает так.
А если не оборачивать, то не даёт скомпилировать.
Вылетает ошибка:

.\main.go:90:12: cannot use time.Since(start).Microseconds() (value of type int64) as time.Duration value in assignment

Reply:
Отмена, нашёл проблему, исправил